tests/integration/golang/aim/namespace/flows/app_test.go

```diff
@@ -0,0 +1,280 @@
+			}()
+
+			// setup namespaces
+			for _, nsCode := range []string{"default", tt.namespace1Code, tt.namespace2Code} {
+				// ignore errors here since default exists on first run
+				//nolint:errcheck
+				_, _ = s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+					Code:                nsCode,
+					DefaultExperimentID: common.GetPointer(int32(0)),
+				})
+			}
+
+			// run actual flow test over the test data.
+			s.testAppFlow(tt.namespace1Code, tt.namespace2Code)
+		})
+	}
+}
+
+func (s *AppFlowTestSuite) testAppFlow(
+	namespace1Code, namespace2Code string,
+) {
+	// create Apps
+	app1ID := s.createApp(namespace1Code, &request.CreateApp{
+		Type: "tf",
+		State: request.AppState{
+			"app-state-key": "app-state-value1",
+		},
+	})
+
+	app2ID := s.createApp(namespace2Code, &request.CreateApp{
+		Type: "mpi",
+		State: request.AppState{
+			"app-state-key": "app-state-value2",
+		},
+	})
+
+	// test `GET /apps` endpoint with namespace 1
+	resp := []response.App{}
+	require.Nil(
+		s.T(),
+		s.AIMClient().WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			"/apps",
+		),
+	)
+	// only app 1 should be present
+	assert.Equal(s.T(), 1, len(resp))
+	assert.Equal(s.T(), app1ID, resp[0].ID)
+
+	// test `GET /apps` endpoint with namespace 2
+	require.Nil(
+		s.T(),
+		s.AIMClient().WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace2Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&resp,
+		).DoRequest(
+			"/apps",
+		),
+	)
+	// only app 2 should be present
+	assert.Equal(s.T(), 1, len(resp))
+	assert.Equal(s.T(), app2ID, resp[0].ID)
+
+	// IDs from other namespace cannot be fetched, updated, or deleted
+	errResp := response.Error{}
+	client := s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodGet,
+		).WithNamespace(
+			namespace1Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&errResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app2ID),
+		),
+	)
+	assert.Equal(s.T(), fiber.ErrNotFound.Code, client.GetStatusCode())
+
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodPut,
+		).WithNamespace(
+			namespace2Code,
+		).WithRequest(
+			request.UpdateApp{
+				Type: "app-type",
+				State: request.AppState{
+					"app-state-key": "new-app-state-value",
+				},
+			},
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&errResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app1ID),
+		),
+	)
+	assert.Equal(s.T(), fiber.ErrNotFound.Code, client.GetStatusCode())
+
+	client = s.AIMClient()
+	require.Nil(
+		s.T(),
+		client.WithMethod(
+			http.MethodDelete,
+		).WithNamespace(
+			namespace2Code,
+		).WithResponseType(
+			helpers.ResponseTypeJSON,
+		).WithResponse(
+			&errResp,
+		).DoRequest(
+			fmt.Sprintf("/apps/%s", app1ID),
```

Comment:
I just  that we don't need to use `fmt.Sprintf`. it will be done under the hood automatically =). I added this moment for just forgot.

Reply:
I think we should stick with what is expected by most programmers (use fmt.Sprintf for template string + values).

Reply:
but you can just pass `DoRequest("/apps/%s", app1ID)` like for example in `errors.Errorf()` function. You don't need to pass `sprintf` here, because `sprintf` is under the hood probably.

Reply:
That's a surprising behavior. Maybe add `DoRequestf()` to indicate templating is available?